lib/commands/gesture.js

```diff
@@ -25,8 +39,199 @@ commands.click = async function (el) {
+  } else if (result.status === youiEngineDriverReturnValues.WEBDRIVER_NO_SUCH_ELEMENT) {
+    throw new errors.NoSuchElementError();
+  } else if (result.status === youiEngineDriverReturnValues.WEBDRIVER_STALE_ELEMENT) {
+    throw new errors.StaleElementReferenceError();
+  }
 
   return result.value;
+};
+
+commands.tapXY = async function(x, y, longPress) {
+
+  // perform a tap at the given x,y screen coordinate
+  // if longPress is true, the tap becomes a longPress action
+
+  let commandName = 'tap';
+  if (longPress === true) {
+    commandName = 'longpress';
+  }
 
+  var commandObject = {
+    name: commandName,
+    args: [`${x}`, `${y}`]
+  };
+
+  var commandJSON = JSON.stringify(commandObject);
+  let data = await this.executeSocketCommand(commandJSON);
+
+  let result;
+  try {
+    result = JSON.parse(data);
+  } catch (e) {
+    throw new Error("Bad response from tapXY");
+  }
+
+  // get status returned
+  if (result.status === youiEngineDriverReturnValues.WEBDRIVER_STALE_ELEMENT)
+    throw new errors.StaleElementReferenceError();
+
+  return result.value;
 };
+
+commands.tap = async function (gestures, longPress) {
+
+  // parse the given gestures array to call the appropriate tap method
+  // if longPress is true, the tap is a long press action
+
+  let elementId = gestures[0].options.element;
+  let x = gestures[0].options.x;
+  let y = gestures[0].options.y;
+
+  if (elementId) {
+
+    // we are either tapping on the default location of the element
+    // top left has been specified of the element has been specified
+
+    if (x && y) {
+      await this.tapElXY(elementId, x, y, longPress);
+    } else {
+      // tapping with an element parameter is the same as click on a WebElement so they can
+      // filter through the same call stack
+      await this.tapEl(elementId, longPress);
+    }
+  } else {
+
+    // we are tapping at an absolute x,y coordinate on the screen
+    await this.tapXY(x, y, longPress);
+  }
+};
+
+commands.doSwipe = async function (gestures, longPress) {
+
+  // parse input values
+  let press = gestures[0];
+  let moveTo = gestures[1];
+  let startX = press.options.x || 0,
+      startY = press.options.y || 0,
+      endX = moveTo.options.x || 0,
+      endY = moveTo.options.y || 0;
+
+  if (press.options.element) {
+    let {x, y} = await this.getLocation(press.options.element);
+    let {width, height} = await this.getSize(press.options.element);
+
+
+    if (!press.options.x && !press.options.y && width && height) {
+      startX += width / 2;
+      startY += height / 2;
+    }
+
+    startX += x || 0;
+    startY += y || 0;
+
+  }
+  if (moveTo.options.element) {
+    let {x, y} = await this.getLocation(moveTo.options.element);
+    let {width, height} = await this.getSize(moveTo.options.element);
+    endX += x || 0;
+    endY += y || 0;
+
+    // let's use the middle of the element to move to, unless an offset from
+    // top left has been specified of the element has been specified
```

Comment:
// let's use the middle of the element to move to, unless an offset from
 // **the** top left ~~has been specified~~ of the element has been specified